storage/badger/badger.go

```diff
@@ -0,0 +1,85 @@
+package badger
+
+import (
+	v3Badger "github.com/dgraph-io/badger/v3"
+	"github.com/pastelnetwork/go-commons/errors"
+	"github.com/pastelnetwork/walletnode/storage"
+
+	"github.com/pastelnetwork/go-commons/log"
+)
+
+// DB structure represents wrapper over the badger.DB
+type DB struct {
+	db     *v3Badger.DB
+	logger *Logger
+}
+
+// NewDB creates an instance of BadgerDB and starts fetching data
+func NewDB(cfg *Config) storage.KeyValue {
+	ch := &DB{
+		db:     &v3Badger.DB{},
+		logger: NewLogger(),
+	}
+	if err := ch.start(cfg); err != nil {
+		ch.logger.Errorf("error caused when trying to start badger db from %v", cfg.Dir)
+		return nil
+	}
```

Comment:
Move this code in `Init()` func, and returns an error instead of log record,
How does a app will know that it must stop everything and exit with status 1